### ISSUE TITLE
[iOS] opencv_dart framework is missing MinimumOSVersion

### ISSUE THREAD

**Issue by bonsoirelliot** [rainyl/opencv_dart#246]
This fails to pass asset validation in AppStore connect:
`2024-09-20 10:59:37.825 *** Error: ERROR: [ContentDelivery.Uploader] Asset validation failed (90530) Invalid MinimumOSVersion. Apps that only support 64-bit devices must specify a deployment target of 8.0 or later. MinimumOSVersion in 'Runner.app/Frameworks/opencv_dart.framework' is ''. (ID: 872a52be-448f-4031-a5cd-e38cc218b50b)`
`2024-09-20 10:59:37.825 *** Error: ERROR: [ContentDelivery.Uploader] Asset validation failed (90360) Missing Info.plist value. A value for the key 'MinimumOSVersion' in bundle Runner.app/Frameworks/opencv_dart.framework is required. (ID: f4b2f07b-46da-4ac5-8076-55e3055657dc)`
`2024-09-20 10:59:37.825 *** Error: ERROR: [ContentDelivery.Uploader] Asset validation failed (90208) Invalid Bundle. The bundle Runner.app/Frameworks/opencv_dart.framework does not support the minimum OS Version specified in the Info.plist. (ID: b265665e-2af3-4e89-b806-2dec52c34872)`

**Comment by frg-x** [comment]
I can confirm the problem. It's the same like I wrote
 @rainyl Please take a look here too [https://github.com/rainyl/opencv_dart/issues/200#issuecomment-2363572016](url)

**Comment by rainyl** [comment]
@bonsoirelliot Thanks for reporting the issue. Since I am not an apple user, I didn't test the condition, I will add `MinimumOSVersion` in to `Info.plist`

**Comment by frg-x** [comment]
What should be minimum iOS version to build with `opencv_dart` successfully?

**Comment by rainyl** [comment]
@frg-x try to change the Info.plist to 
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>AvailableLibraries</key>
	<array>
		<dict>
			<key>BinaryPath</key>
			<string>opencv_dart.framework/opencv_dart</string>
			<key>LibraryIdentifier</key>
			<string>ios-arm64</string>
			<key>LibraryPath</key>
			<string>opencv_dart.framework</string>
			<key>SupportedArchitectures</key>
			<array>
				<string>arm64</string>
			</array>
			<key>SupportedPlatform</key>
			<string>ios</string>
			<key>MinimumOSVersion</key>
			<string>12.0</string>
		</dict>
		<dict>
			<key>BinaryPath</key>
			<string>opencv_dart.framework/opencv_dart</string>
			<key>LibraryIdentifier</key>
			<string>ios-x86_64-simulator</string>
			<key>LibraryPath</key>
			<string>opencv_dart.framework</string>
			<key>SupportedArchitectures</key>
			<array>
				<string>x86_64</string>
			</array>
			<key>SupportedPlatform</key>
			<string>ios</string>
			<key>SupportedPlatformVariant</key>
			<string>simulator</string>
			<key>MinimumOSVersion</key>
			<string>12.0</string>
		</dict>
	</array>
	<key>CFBundlePackageType</key>
	<string>XFWK</string>
	<key>XCFrameworkFormatVersion</key>
	<string>1.0</string>
</dict>
</plist>
```

**Comment by frg-x** [comment]
Now instead of 3 errors I have this one:
![screenshot-20-09-2024-15-04-01](https://github.com/user-attachments/assets/9c4d02af-cedc-4cd2-a8c2-69aedb9575af)
Is this path to `Info.plist` correct?
![screenshot-20-09-2024-15-04-45](https://github.com/user-attachments/assets/03b25832-e27c-43f9-a4d8-2609bfb826e1)
Also I didn't notice the moment when application build time increased from 3 minutes to 10 mins. Could it be related to changes that I made to build app with `opencv_dart` package for AppStore?

**Comment by rainyl** [comment]
> Now instead of 3 errors I have this one: ![screenshot-20-09-2024-15-04-01](https://private-user-images.githubusercontent.com/36096131/369414279-9c4d02af-cedc-4cd2-a8c2-69aedb9575af.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY4NDA5MTgsIm5iZiI6MTcyNjg0MDYxOCwicGF0aCI6Ii8zNjA5NjEzMS8zNjk0MTQyNzktOWM0ZDAyYWYtY2VkYy00Y2QyLWE4YzItNjlhZWRiOTU3NWFmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIwVDEzNTY1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI3ZDFiNjg0NTBlMTg2NDQ5ODQ1ZTI4ZDcyYjhlNzM1NTJlYmQxNWQyZWYyNGQ3N2MyMDUzYjU0OWE1NDQ0ZDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ktzlnfB-HrEKkBCeUkodL4BPgExsJCcChTIhrxoBXHw) Is this path to `Info.plist` correct? ![screenshot-20-09-2024-15-04-45](https://private-user-images.githubusercontent.com/36096131/369414567-03b25832-e27c-43f9-a4d8-2609bfb826e1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY4NDA5MTgsIm5iZiI6MTcyNjg0MDYxOCwicGF0aCI6Ii8zNjA5NjEzMS8zNjk0MTQ1NjctMDNiMjU4MzItZTI3Yy00M2Y5LWE0ZDgtMjYwOWJmYjgyNmUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIwVDEzNTY1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAyNmJhZDMzZjNiMjE4Y2Y1MmI0ZGExNzY5NTJhODRjMWU0MTZiMmQ1MDFhYmY4NTc4NWY3ODhkOGRhNjY5MWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ZFJM0bEgLViF5oL4KupWfJu3eKlxQ_-p8B_YcBruQXg) Also I didn't notice the moment when application build time increased from 3 minutes to 10 mins. Could it be related to changes that I made to build app with `opencv_dart` package for AppStore? BTW, thank you @rainyl for soon answers! 🔥

Well.... the location is correct. but it seems not working..

what about add this 
```xml
<key>MinimumOSVersion</key>
<string>12.0</string>
```
to `opencv_dart.xcframwork/ios-arm64/Info.plist` and `opencv_dart.xcframwork/ios-x86_64-simulator/Info.plist`?

**Comment by frg-x** [comment]
After adding suggested strings to the both of `Info.plist` I still have the same issue.
![screenshot-20-09-2024-16-27-46](https://github.com/user-attachments/assets/30a76d8d-1e6f-451d-bc8a-f7dc2eca93ef)
Also I noticed that size of *.ipa is near to 60Mb, though Runner.xcarchive is 370Mb. Maybe *.ipa doesn't include opencv libs or something like that?

**Comment by rainyl** [comment]
@frg-x which version of XCode do you use? I just searched it and there are many similar reports about XCode 15.3 with same error, I will try to fix it but before that you can try to downgrage XCode to 15.2

Or if you do not want to downgrade, try to change 
```xml
<key>MinimumOSVersion</key>
<string>12.0</string>
```

to 

```xml
<key>MinimumOSVersion</key>
<string>100.0</string>
```

as firebase fixed the issue by adding this. I am not an apple developer so have no developer account, building to ipa is not easy for me, what I can do is to fix it according to your response, hope you won't be annoying.

- https://github.com/firebase/firebase-ios-sdk/issues/12441#issuecomment-1981705679
- https://github.com/firebase/firebase-ios-sdk/blob/3ae46c051533cdf2632e4fa6b332df5a4e075c6a/ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift#L577

> Also I noticed that size of *.ipa is near to 60Mb, though Runner.xcarchive is 370Mb. Maybe *.ipa doesn't contain opencv related files or something like that?

maybe some symbols were stripped

**Comment by frg-x** [comment]
I use XCode 15.4, Sonoma 14.7. 
Finally it has built as sent successfully to the AppStore with:
```
<key>MinimumOSVersion</key>
<string>100.0</string>
```

**Comment by rainyl** [comment]
> I use XCode 15.4, Sonoma 14.7. Finally it has built as sent successfully to the AppStore with in all (three) of `Info.plist` files inside `ios/.symlinks/plugins/opencv_dart/ios/opencv_dart.xcframework` folder.
> 
> ```
> <key>MinimumOSVersion</key>
> <string>100.0</string>
> ```
> 
> Thanks a lot @rainyl 🔥🙌

Great, now I know how to fix it :)

**Comment by frg-x** [comment]
That's the advantage of a community😉

**Comment by bonsoirelliot** [comment]
@rainyl hello again :) i've added a package this way, but i got same error
![изображение](https://github.com/user-attachments/assets/cb797edc-341e-4941-8681-7f3428944233)

**Comment by rainyl** [comment]
> @rainyl hello again :) i've added a package this way, but i got same error ![изображение](https://private-user-images.githubusercontent.com/27853419/369761558-cb797edc-341e-4941-8681-7f3428944233.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcwNzUxMDYsIm5iZiI6MTcyNzA3NDgwNiwicGF0aCI6Ii8yNzg1MzQxOS8zNjk3NjE1NTgtY2I3OTdlZGMtMzQxZS00OTQxLTg2ODEtN2YzNDI4OTQ0MjMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIzVDA3MDAwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMzYmIwZDIwMGMzNzFhMWM0MDA2YTM3ZTkzZDM4YzUwNjU2ODk2MWVmNWVhZWYzNTIxMGQxZjZkNWY4NTdiN2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.l1wODzuaCmKi2_5k3srZmT-W4B262fh6VhikJpG4wwM)

@bonsoirelliot This doesn't work because new binary libs need to be built.

I am planning publish v1.2.5, but with some extra things todo, you can download them from https://github.com/rainyl/opencv_dart/actions/runs/10968554580 for now.

**Comment by frg-x** [comment]
@bonsoirelliot I've managed to add package by this code:
```
  opencv_dart:
    git: 
      url: https://github.com/rainyl/opencv_dart.git
      ref: '6384668'
```
![screenshot-23-09-2024-09-05-46](https://github.com/user-attachments/assets/85bb10b1-880c-4a46-b598-dc4761c93d06)

**Comment by rainyl** [comment]
> @bonsoirelliot I've managed to add package by this code:
> 
> ```
>   opencv_dart:
>     git: 
>       url: https://github.com/rainyl/opencv_dart.git
>       ref: '6384668'
> ```
> 
> ![screenshot-23-09-2024-09-05-46](https://private-user-images.githubusercontent.com/36096131/369777714-85bb10b1-880c-4a46-b598-dc4761c93d06.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcwNzU3NDMsIm5iZiI6MTcyNzA3NTQ0MywicGF0aCI6Ii8zNjA5NjEzMS8zNjk3Nzc3MTQtODViYjEwYjEtODgwYy00YTQ2LWI1OTgtZGM0NzYxYzkzZDA2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIzVDA3MTA0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFmZTYwMWMxNTQ1YmRmMjY5YjY5ZTM1ZjA4YzNiNmJjZjMwYjYzY2JiZWQwYjA5NGU2MjcwZTAxMTg0ZjYyODcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xqe-eksQe7h66ClAmPhsjRnkWGqR473RbpMG4WQ9goA)
> 
> `pod install` ![screenshot-23-09-2024-09-09-03](https://private-user-images.githubusercontent.com/36096131/369778441-c12f2623-a84c-40ee-bc02-03ba8d65f2ce.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcwNzU3NDMsIm5iZiI6MTcyNzA3NTQ0MywicGF0aCI6Ii8zNjA5NjEzMS8zNjk3Nzg0NDEtYzEyZjI2MjMtYTg0Yy00MGVlLWJjMDItMDNiYThkNjVmMmNlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIzVDA3MTA0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFkOTZhNjMyNWVhYWYxZmE5NWM4NmEwNmIzOTE2M2M2YmExZGYwYjk1YTliYjI0NDAxNzlkYzU2MDM3NWY4ZTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.TN46p6l6Rerp_gvfVf5jzxtWRE-f_rpSMXyrgIwYCZg)

emmmmm, I am sorry but theoretically this also doesn't work, the downloaded binaries are still for v1.2.4 with incorrect `Info.plist`.

@abdelaziz-mahdy Do you think that we should make the setup scripts to build the libs locally?

**Comment by frg-x** [comment]
But also only `ios/.symlinks/plugins/opencv_dart/ios/Info.plist.in` contains 
```
    <key>MinimumOSVersion</key>
    <string>100.0</string>
```
There is no such strings in any of three `Info.plist` files in `opencv_dart.xcframework` folder and subfolders

**Comment by abdelaziz-mahdy** [comment]
> > @bonsoirelliot I've managed to add package by this code:
> > 
> > ```
> >   opencv_dart:
> >     git: 
> >       url: https://github.com/rainyl/opencv_dart.git
> >       ref: '6384668'
> > ```
> > 
> > ![screenshot-23-09-2024-09-05-46](https://private-user-images.githubusercontent.com/36096131/369777714-85bb10b1-880c-4a46-b598-dc4761c93d06.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcwNzU3NDMsIm5iZiI6MTcyNzA3NTQ0MywicGF0aCI6Ii8zNjA5NjEzMS8zNjk3Nzc3MTQtODViYjEwYjEtODgwYy00YTQ2LWI1OTgtZGM0NzYxYzkzZDA2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIzVDA3MTA0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFmZTYwMWMxNTQ1YmRmMjY5YjY5ZTM1ZjA4YzNiNmJjZjMwYjYzY2JiZWQwYjA5NGU2MjcwZTAxMTg0ZjYyODcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xqe-eksQe7h66ClAmPhsjRnkWGqR473RbpMG4WQ9goA)
> > 
> > `pod install` ![screenshot-23-09-2024-09-09-03](https://private-user-images.githubusercontent.com/36096131/369778441-c12f2623-a84c-40ee-bc02-03ba8d65f2ce.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcwNzU3NDMsIm5iZiI6MTcyNzA3NTQ0MywicGF0aCI6Ii8zNjA5NjEzMS8zNjk3Nzg0NDEtYzEyZjI2MjMtYTg0Yy00MGVlLWJjMDItMDNiYThkNjVmMmNlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIzVDA3MTA0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFkOTZhNjMyNWVhYWYxZmE5NWM4NmEwNmIzOTE2M2M2YmExZGYwYjk1YTliYjI0NDAxNzlkYzU2MDM3NWY4ZTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.TN46p6l6Rerp_gvfVf5jzxtWRE-f_rpSMXyrgIwYCZg)
> 
> emmmmm, I am sorry but theoretically this also doesn't work, the downloaded binaries are still for v1.2.4 with incorrect `Info.plist`.
> 
> @abdelaziz-mahdy Do you think that we should make the setup scripts to build the libs locally?

I don't mind the local build but some people will not have the needed deps to build so it will be harder to manage for people, why do we need to have a local build?

**Comment by rainyl** [comment]
> I don't mind the local build but some people will not have the needed deps to build so it will be harder to manage for people, why do we need to have a local build?
@abdelaziz-mahdy you are right, it just because users can't use it without new binaries built.

@bonsoirelliot @frg-x So plese wait for a new release. With some extra issues solved, v1.2.5 will be published.

**Comment by abdelaziz-mahdy** [comment]
> > I don't mind the local build but some people will not have the needed deps to build so it will be harder to manage for people, why do we need to have a local build?
> 
> @abdelaziz-mahdy you are right, it just because users can't use it without new binaries built.
> 
> @bonsoirelliot @frg-x So plese wait for a new release. With some extra issues solved, v1.2.5 will be published.

Then we can make the setup have a local build by env or a command line argument, so for the cases that need local build it will be there what do you think?

**Comment by rainyl** [comment]
> Then we can make the setup have a local build by env or a command line argument, so for the cases that need local build it will be there what do you think?

Well, I prefer to let the users build by themselves.

**Comment by abdelaziz-mahdy** [comment]
> > Then we can make the setup have a local build by env or a command line argument, so for the cases that need local build it will be there what do you think?
> 
> Well, I prefer to let the users build by themselves.

I meant we add a way to make the setup command go through the build process instead of downloading the libs.

**Comment by bonsoirelliot** [comment]
> > I don't mind the local build but some people will not have the needed deps to build so it will be harder to manage for people, why do we need to have a local build?
> 
> @abdelaziz-mahdy you are right, it just because users can't use it without new binaries built.
> 
> @bonsoirelliot @frg-x So plese wait for a new release. With some extra issues solved, v1.2.5 will be published.

when will v1.2.5 be published?

**Comment by rainyl** [comment]
@bonsoirelliot I am working on supporting ffmpeg on all platforms without the dependency of system packages, so I prefer to publish v1.2.5 with this feature.

If you need it urgently, I can publish it with the current state.

**Comment by rainyl** [comment]
v1.2.5 has been published, now you can upgrade to v1.2.5 and try whether the problem has been solved, if you still have any problems, please open new issues.
